Support systemd-portabled

phaer opened the pull request:
###### Description of changes

Since @gdamjan recently added [tooling](https://github.com/NixOS/nixpkgs/commit/499aebcf340f48ef02211700b39512c429bf88bb) to create portable services for systemd, It'd be nice to be able to run those services on NixOS as well.

We currently build systemd with `withPortabled = false` by default, and even if you enable it there are a few files missing.
This PR adds missing files, sets `withPortabled = true` by default and includes a basic NixOS test to check whether a simple portable service can be attached and detached again. 
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

phaer committed:
systemd: build with portabled by default.

phaer committed:
nixos/tests: init systemd-portabled

 Please enter the commit message for your changes. Lines starting

SuperSandro2000 commented on nixos/modules/system/boot/systemd.nix:
```diff
@@ -151,7 +151,11 @@ let
+    ] ++ optionals cfg.package.withPortabled [
+      "dbus-org.freedesktop.portable1.service"
+      "systemd-portabled.service"
+    ]
+    ++ [
```
```suggestion
    ] ++ [
```

SuperSandro2000 commented on nixos/tests/systemd-portabled.nix:
```diff
@@ -0,0 +1,56 @@
+import ./make-test-python.nix ({pkgs, lib, ...}: let
+  systemdWithPortabled = pkgs.systemd.override {
+    withPortabled = true;
+  };
```
```suggestion
```
This is not really required since portabled is on by default.

phaer replied:
Ah yes, well it wasn't at the start of this PR.  Didn't remove it when I set the default to true, because I wanted to find out how ofborg reacts to a change needing sooo many rebuilds :)

SuperSandro2000 reviewed (approved):
This PR rebuilds a lot of packages which means we must target staging. Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging) to not potentially ping a lot of people.

flokli commented:
> Please note that this is a draft, as the test is able to "attach" and run a portable service, but I did not manage to detach it so far.

Should we mark this PR as draft, until this is figured out, or still disable it by default?

I'd like to avoid people getting their hopes high, trying it out, and then not being able to detach stuff.

gdamjan commented:
> Should we mark this PR as draft, until this is figured out, or still disable it by default?
> 
> I'd like to avoid people getting their hopes high, trying it out, and then not being able to detach stuff.

that was solved
detach needs to mirror the attach options wrt --runtime / --now / --enable

phaer commented:
> Should we mark this PR as draft, until this is figured out, or still disable it by default?
As @gdamjan just wrote, we'd solved that yesterday. I undrafted the PR when the detach test started working, but forgot to update the description. Sorry.


@SuperSandro2000 Thanks for the hint, I squashed some commits and rebased, so the PR targets staging now and I believe we've avoided yet another mass-notification so far :sweat_smile:

flokli commented on nixos/tests/systemd-portabled.nix:
```diff
@@ -0,0 +1,51 @@
+import ./make-test-python.nix ({pkgs, lib, ...}: let
```
Can you add this to the nixosTests attribute set, too?

phaer replied:
Ah, hadn't seen that one before tbh, was looking for a `default.nix` but didn't see `all-tests.nix` up until now.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/625
